nevergrad/functions/causaldiscovery/test_core.py

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import nevergrad as ng
+from . import core
+
+
+def test_causal_discovery_using_data() -> None:
+    func = core.CausalDiscovery(generator='sachs')
+    assert func._nvars == 11
+
+
+def test_causal_discovery_using_generator() -> None:
+    nnodes = 13
+    npoints = 55
+    func = core.CausalDiscovery(generator='acylicgraph', npoints=npoints, nodes=nnodes)
+    assert func._nvars == nnodes
+    assert func._data.shape == (npoints, nnodes)
+    assert func.graph_score(func._ground_truth_graph) == 1
+    optimizer = ng.optimizers.NGO(parametrization=func.parametrization, budget=500)
+    recommendation = optimizer.minimize(func.function)
```

Comment:
this should be minimizing func, not func.function

Reply:
if it was not for the check of min score so far below(which is only for debugging?), I would have said that there is no need to check that we can perform optimization here.
When implementing functions, I would rather verify that the output of an evaluation is correct, aka test an actual output of func directly.
This is particularly useful to avoid regressions, because this way we can make sure the function does not change, since we have a tests that verifies against an actual value.

Can you please add such a test?

Reply:
Added `assert math.isclose(17.425619834710744, result, abs_tol=1e-10)` in `test_causal_discovery_using_data` as the graph will not change in this case.